src/slurm_plugin/clustermgtd.py

```diff
@@ -964,6 +989,93 @@ def _is_node_in_replacement_valid(self, node, check_node_is_valid):
             return not time_is_expired if check_node_is_valid else time_is_expired
         return False
 
+    def _handle_ice_nodes(self, active_nodes: list, unhealthy_dynamic_nodes: list):
```

Comment:
what if any of the methods in here raises an exception? do we need to add an `log_exception` annotation so that we don't break other functionalities that might happen after `_handle_ice_nodes`?

Reply:
Yes, it will be better add `log_exception`. Added